src/discv5.rs

```diff
@@ -479,6 +488,93 @@ impl Discv5 {
         }
     }
 
+    // Use find_topic to find the Enrs the shortest XOR distance from the topic hash,
+    // and send the topic query to these nodes
+    pub fn topic_query_req(
+        &self,
+        enr: Enr,
+        topic_hash: TopicHash,
+    ) -> impl Future<Output = Result<Vec<Enr>, RequestError>> + 'static {
+        // convert the ENR to a node_contact.
+        let node_contact = NodeContact::from(enr);
+
+        // the service will verify if this node is contactable, we just send it and
+        // await a response.
+        let (callback_send, callback_recv) = oneshot::channel();
+        let channel = self.clone_channel();
+
+        async move {
+            let channel = channel.map_err(|_| RequestError::ServiceNotStarted)?;
+
+            let event = ServiceRequest::TopicQuery(node_contact, topic_hash, callback_send);
+
+            // send the request
+            channel
+                .send(event)
+                .await
+                .map_err(|_| RequestError::ChannelFailed("Service channel closed".into()))?;
+            // await the response
+            callback_recv
+                .await
+                .map_err(|e| RequestError::ChannelFailed(e.to_string()))?
+        }
+    }
+
+    // Use find_topic to find the Enrs the shortest XOR distance from the topic hash,
+    // and send the topic query to these nodes
+    pub fn reg_topic_req(
+        &self,
+        enr: Enr,
```

Comment:
Same here?

Reply:
No, here too the idea is to find out which nodes to send the requests to in a separate call to the discv5 instance. I did it like this for simplicity of doing the first integration testing. https://github.com/emhane/discv5-cli/blob/2318f77f4017b2f9f35b5f91be11628c8e66c462/src/topic/mod.rs#L210